pkg/tfbridge/provider.go

```diff
@@ -588,41 +588,74 @@ func (p *Provider) typeCheckConfig(
 	// for now we are just going to log warnings if there are failures.
 	// over time we may want to turn these into actual errors
 	validateShouldError := cmdutil.IsTruthy(os.Getenv("PULUMI_ERROR_CONFIG_TYPE_CHECKER"))
+
+	// If we don't have a schema, then we don't attempt to type check the config at
+	// all.
+	if p.pulumiSchemaSpec == nil {
+		logger.Debug("p.pulumiSchemaSpec == nil, skipping type checking config")
+		return nil
+	}
+
+	iv := NewInputValidator(urn, *p.pulumiSchemaSpec, true)
+	typeFailures := iv.ValidateConfig(news)
+	if validateShouldError {
+		return p.convertTypeFailures(urn, typeFailures)
+	}
+
+	// If we don't have any type errors, we can just return.
+	if len(typeFailures) == 0 {
+		return nil
+	}
+
+	// warningSpaces is same length as the string that prefixes a [logger.Warn] invocation.
+	//
+	//                    "warning: "
+	const warningSpaces = "         "
```

Comment:
Is there a reason tabs won't work here and for `fourSpaces`? I know we can't get tabs to be the same exact spacing as `warning: ` but they're more visible when reading this code.

Reply:
Tabs didn't display the same way for me in my in-editor (emacs/vterm) terminal and my separate terminal (Alacritty).